audit_engine/dynamic_analysis/__init__.py

```
@@ -344,47 +423,79 @@ def run_dynamic_analysis(
+def _run_async_analysis(
+    orchestrator: DynamicAnalysisOrchestrator, 
+    contract_paths: Iterable[str]
+) -> List[AnalysisResult]:
+    """Run async analysis with proper event loop handling"""
     try:
-        asyncio.get_running_loop()
-    except RuntimeError:
-        return asyncio.run(orchestrator.analyze_contracts(contract_paths))
-    else:
+        # Check if we're already in an event loop
+        loop = asyncio.get_running_loop()
```

Ruff (0.12.2)

Local variable `loop` is assigned to but never used

Remove assignment to unused variable `loop`

(F841)